src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -258,7 +258,23 @@ private static boolean shouldBeDeclaredAsFinal(ClassDesc desc) {
                     || desc.isSuperClassOfAnonymousInnerClass())
                 && !desc.isDeclaredAsFinal()
                 && !desc.isWithNonPrivateCtor()
-                && !desc.isWithNestedSubclass();
+                && !desc.isWithNestedSubclass()
+                || isPrivateWithDefaultCtor(desc, desc.getTypeDeclarationAst());
```

Comment:
indent `||` +4 spaces to show that it is not same priority of execution as `&&`.

make single arguemnt for this method.

Reply:
done